index.js

```diff
@@ -5,40 +5,84 @@ var objectAssign = require('object-assign');
 var FormData = require('form-data');
 var stream = require('stream');
 var https = require('https');
+var request = require('request');
+var async = require('async');
+var fileType = require('file-type');
 
 var host = 'news-api.apple.com';
 var CRLF = '\r\n';
 
+var validContentTypes = [
+  'application/octet-stream', // should be first in this list
+  'image/jpeg',
+  'image/png',
+  'image/gif'
+];
+
 function encodeFormData (formData, cb) {
   var form = new FormData();
 
-  Object.keys(formData).forEach(function (item) {
-    var value = formData[item].value;
-    var options = formData[item].options;
-    form.append(item, value, objectAssign({
-      header: '--' + form.getBoundary() + CRLF +
-        'Content-Type: application/json' + CRLF +
-        'Content-Disposition: form-data; filename=' + options.filename +
-          '; name=' + item +
-          '; size=' + Buffer.byteLength(value) + CRLF + CRLF
-    }, formData[item].options));
-  });
+  async.mapSeries(formData, function (item, callback) {
+    var value = item.value;
+    var options = item.options;
+
+    function append () {
+      var len = typeof value === 'string' ? Buffer.byteLength(value) : value.length;
+      var header = '--' + form.getBoundary() + CRLF +
+        'Content-Type: ' + options.contentType + CRLF +
+        'Content-Disposition: form-data' +
+        (item.filename ? '; filename=' + encodeURIComponent(options.filename) : '') +
+        (item.name ? '; name=' + encodeURIComponent(item.name) : '') +
+        '; size=' + len + CRLF + CRLF;
+
+      form.append(item.name || item.filename, value,
+        objectAssign({ header: header, knownLength: len }, options));
+      callback();
+    }
 
-  var converter = new stream.Writable();
-  var result = '';
-  converter._write = function (chunk, enc, cb) {
-    result += chunk.toString();
-    cb();
-  };
+    if (options.contentType === 'application/json') {
+      return append();
+    }
+
+    request.get(value, {
+      timeout: 5000,
+      encoding: null,
+      rejectUnauthorized: process.env.NODE_ENV !== 'test'
+    }, function (err, res, body) {
```

Comment:
@iefserge might be worth taking a look at https://github.com/ShogunPanda/fastimage for this?